server/plugin/webhook.go

```diff
@@ -92,47 +92,79 @@ func (p *Plugin) handleWebhook(w http.ResponseWriter, r *http.Request) {
 	switch event := event.(type) {
 	case *github.PullRequestEvent:
 		repo = event.GetRepo()
+		var shouldNotNotify, err = p.CheckIsNotificationOff(*repo.FullName)
```

Comment:
Are there cases where `CheckIsNotificationOff` should not get checked? If not, can the check the unified?

Reply:
Sorry , i didn't got this 
currently for every event handled in webhook it is configured

Reply:
Right now the code checks `CheckIsNotificationOff` in every `case`. Can this be unified to only check once?

Reply:
seems the event is a interface , so need a type switch to get the repository name from event so added in the existing switch only , let me know if we can do it in better way

Reply:
Can you put the error checking into the `CheckIsNotificationOff` function and remove the redundant code?  Maybe rename the function also..  Each `case` could just have something like the following:

```go
	case *github.PullRequestEvent:
		repo = event.GetRepo()
		if p.IsNotificationOff(*repo.Fullname) {
        	return
		}
		handler = func() {
		  // do handling
		}
```

Reply:
done